### ISSUE TITLE
eul status bar components are force hidden by system

### ISSUE THREAD

**Issue by zhqu1148980644** [gao-sun/eul#149]
<h3>Make sure there's no open issue for the same bug before submit.</h3>

**Describe the Bug**

eul status bar components are force hidden by system

**Expected Behavior**

see the eul status bar

**Screenshots**

![image](https://user-images.githubusercontent.com/29366130/105354167-22ac5300-5c2b-11eb-8e1e-6ed6fd6b3a99.png)

**Context**
 - eul version:
 - macOS version:
 - Device model: 

**Debug Output**

``` bash
# PASTE OUTPUT HERE #
```

**Is Related to a Crash?**

**Comment by jevonmao** [comment]
The status bar has limited amount of space. It's not a bug, try taking off other icons on status bar or use less number of components in eul preferences

**Comment by CedricRBR** [comment]
I get the same error, but all components are shown. So it's not a space issue.
It looks like eul doesn't like it when the menu bar is set to auto hide and the settings pane is open.

**Comment by gao-sun** [comment]
@CedricRBR didn't consider the auto hide setting, thank you for reporting! will take a look soon

**Comment by TinQ0** [comment]
I have the same issue first time launching eul when bartender is enabled.

**Comment by gao-sun** [comment]
currently I'm using a delay detection of visibility (~15s) after booting up and actually I noticed it's still too short under some circumstances. it's kind of tricky to know whether the system is "ready" for displaying status bar items or it's trying to hide status bar items due to the lack of space. I'll try to figure this out as well.

**Comment by jasikpark** [comment]
Would it be possible to split up the menubar into individual bits, so that it isn't all or nothing? So that if you have a particularly large window menu, your fan and network might be covered up, but you can still see your cpu usage?

**Comment by gao-sun** [comment]
@jasikpark yes, this should be a reasonable feature as the number of components grow. need some time to think about the implementation.

**Comment by RainThinks** [comment]
> > I have the same issue first time launching eul when bartender is enabled.
> 
> Similar issue here.
> 
> 1. Position eul outside of [Hidden](https://github.com/dwarvesf/hidden) i.e. always visible
> 2. Reboot machine
> 3. eul has moved into the _hidden_ section of Hidden, and this alert shows:
>    ![screenshot](https://camo.githubusercontent.com/c4816dfad682226e2f009ca1f33d53b676c523df7a4fd32488068e2f566d1d9e/68747470733a2f2f692e696d6775722e636f6d2f4559584b4933782e706e67)

same here.

**Comment by RainThinks** [comment]
> > I have the same issue first time launching eul when bartender is enabled.
> 
> Similar issue here.
> 
> 1. Position eul outside of [Hidden](https://github.com/dwarvesf/hidden) i.e. always visible
> 2. Reboot machine
> 3. eul has moved into the _hidden_ section of Hidden, and this alert shows:
>    ![screenshot](https://camo.githubusercontent.com/c4816dfad682226e2f009ca1f33d53b676c523df7a4fd32488068e2f566d1d9e/68747470733a2f2f692e696d6775722e636f6d2f4559584b4933782e706e67)

Same here.

I'm only enabled network widget in menu, still force hidden by system.

**Comment by mirajjiwani** [comment]
I had the same issue but it was because the EUL components were "Always Hidden" in my Ice App (alternative to Bartender). Once I moved them to the Hidden or Always section in Ice, it showed up without error.
